File: scripts/fix_headers.py
"""
manager_ga 파일 헤더 정리 스크립트
- "주요 기능:" 섹션 제거
- "사용 위치:" → "사용 페이지:" 통일
- 헤더 이모지(📊📋📅 등) 제거
"""
import re
import os

files = [
    # components/manager/ga
    "src/components/manager/ga/campaign/rejected/section/CampaignRejectedFilterSection.tsx",
    "src/components/manager/ga/campaign/rejected/section/RejectCodeInfoSection.tsx",
    "src/components/manager/ga/campaign/rejected/section/RejectStatsSection.tsx",
    "src/components/manager/ga/campaign/reported/section/CampaignReportedFilterSection.tsx",
    "src/components/manager/ga/campaign/reported/section/ReportCodeInfoSection.tsx",
    "src/components/manager/ga/campaign/reported/section/ReportStatsSection.tsx",
    "src/components/manager/ga/common/ManagerGAHeader.tsx",
    "src/components/manager/ga/common/SidebarMenu.tsx",
    "src/components/manager/ga/common/filter/BaseFilterDropdown.tsx",
    "src/components/manager/ga/common/filter/BaseFilterModal.tsx",
    "src/components/manager/ga/common/filter/BaseFilterSection.tsx",
    "src/components/manager/ga/common/filter/DateFilterButton.tsx",
    "src/components/manager/ga/common/filter/FilterButton.tsx",
    "src/components/manager/ga/common/filter/SortDropdown.tsx",
    "src/components/manager/ga/dashboard/ChartsSection.tsx",
    "src/components/manager/ga/dashboard/MemberStatsSection.tsx",
    "src/components/manager/ga/dashboard/StatCard.tsx",
    "src/components/manager/ga/dashboard/chart/CampaignRecruitmentChart.tsx",
    "src/components/manager/ga/dashboard/chart/ChannelMemberPieChart.tsx",
    "src/components/manager/ga/dashboard/chart/DeviceStatsChart.tsx",
    "src/components/manager/ga/dashboard/chart/MemberActivationDonutChart.tsx",
    "src/components/manager/ga/dashboard/chart/MemberTypeBarChart.tsx",
    "src/components/manager/ga/dashboard/chart/RejectionReportChart.tsx",
    "src/components/manager/ga/dashboard/section/AccessStatsSection.tsx",
    "src/components/manager/ga/dashboard/section/CampaignRecruitmentSection.tsx",
    "src/components/manager/ga/dashboard/section/CampaignSummarySection.tsx",
    "src/components/manager/ga/dashboard/section/ChannelMemberSection.tsx",
    "src/components/manager/ga/dashboard/section/DateFilterSection.tsx",
    "src/components/manager/ga/dashboard/section/DateRangePickerModal.tsx",
    "src/components/manager/ga/dashboard/section/MemberActivationSection.tsx",
    "src/components/manager/ga/dashboard/section/MemberTypeSection.tsx",
    "src/components/manager/ga/dashboard/section/RejectionReportSection.tsx",
    # data/manager_ga
    "src/data/manager_ga/common/filterOptions.ts",
    "src/data/manager_ga/community/categoriesData.ts",
    "src/data/manager_ga/community/postsData.ts",
    "src/data/manager_ga/member/blacklist.ts",
    "src/data/manager_ga/member/partners.ts",
    "src/data/manager_ga/member/reviewers.ts",
    "src/data/manager_ga/progress.ts",
    "src/data/manager_ga/rejected.ts",
    "src/data/manager_ga/reported.ts",
]

# 이모지 제거 패턴 (헤더 구분선 내의 이모지)
EMOJI_PATTERN = re.compile(
    r'(\/\* ={38,}\n\s+)'  # /* ====... 시작
    r'([^\n]*)',            # 제목 라인
    re.MULTILINE
)

def remove_emoji(text):
    """유니코드 이모지 제거"""
    emoji_re = re.compile(
        "["
        "\U0001F300-\U0001F5FF"
        "\U0001F600-\U0001F64F"
        "\U0001F680-\U0001F6FF"
        "\U0001F700-\U0001F77F"
        "\U0001F780-\U0001F7FF"
        "\U0001F800-\U0001F8FF"
        "\U0001F900-\U0001F9FF"
        "\U0001FA00-\U0001FA6F"
        "\U0001FA70-\U0001FAFF"
        "\U00002702-\U000027B0"
        "\U000024C2-\U0001F251"
        "]+", flags=re.UNICODE
    )
    return emoji_re.sub('', text)

def fix_header(content):
    # 1. "주요 기능:" 섹션 제거
    # 패턴: " * 주요 기능:\n" 이후 " * - ..." 라인들과 빈 " *\n" 라인까지
    content = re.sub(
        r' \* 주요 기능:\n( \* - [^\n]*\n)* \*\n',
        '',
        content
    )
    # 주요 기능 섹션 뒤에 빈줄이 없는 경우도 처리
    content = re.sub(
        r' \* 주요 기능:\n( \* - [^\n]*\n)+',
        '',
        content
    )

    # 2. "사용 위치:" → "사용 페이지:" 통일
    content = content.replace(' * 사용 위치:\n', ' * 사용 페이지:\n')

    # 3. 헤더 구분선 내 이모지 제거 (/* === 제목 === */ 부분)
    def clean_header_line(m):
        prefix = m.group(1)
        title = m.group(2)
        clean_title = remove_emoji(title).strip()
        return prefix + clean_title

    content = EMOJI_PATTERN.sub(clean_header_line, content)

    # 4. 연속 빈 줄 정리 (/** 블록 내에서 3줄 이상 빈 줄 → 1줄)
    content = re.sub(r'( \*\n){3,}', ' *\n', content)

    return content

base = "C:/develop/reviewx-web"
modified = 0
errors = []

for rel_path in files:
    full_path = os.path.join(base, rel_path)
    try:
        with open(full_path, 'r', encoding='utf-8') as f:
            original = f.read()

        fixed = fix_header(original)

        if fixed != original:
            with open(full_path, 'w', encoding='utf-8', newline='\n') as f:
                f.write(fixed)
            modified += 1
            print(f"✅ 수정: {rel_path}")
        else:
            print(f"⏭️  변경 없음: {rel_path}")
    except FileNotFoundError:
        errors.append(f"❌ 파일 없음: {rel_path}")
    except Exception as e:
        errors.append(f"❌ 오류 ({rel_path}): {e}")

print(f"\n총 {modified}개 파일 수정 완료")
if errors:
    print("\n오류 목록:")
    for e in errors:
        print(e)

File: scripts/fix_headers.mjs
import fs from 'fs';
import path from 'path';

const base = 'C:/develop/reviewx-web';

const files = [
  // components/manager/ga
  'src/components/manager/ga/campaign/rejected/section/CampaignRejectedFilterSection.tsx',
  'src/components/manager/ga/campaign/rejected/section/RejectCodeInfoSection.tsx',
  'src/components/manager/ga/campaign/rejected/section/RejectStatsSection.tsx',
  'src/components/manager/ga/campaign/reported/section/CampaignReportedFilterSection.tsx',
  'src/components/manager/ga/campaign/reported/section/ReportCodeInfoSection.tsx',
  'src/components/manager/ga/campaign/reported/section/ReportStatsSection.tsx',
  'src/components/manager/ga/common/ManagerGAHeader.tsx',
  'src/components/manager/ga/common/SidebarMenu.tsx',
  'src/components/manager/ga/common/filter/BaseFilterDropdown.tsx',
  'src/components/manager/ga/common/filter/BaseFilterModal.tsx',
  'src/components/manager/ga/common/filter/BaseFilterSection.tsx',
  'src/components/manager/ga/common/filter/DateFilterButton.tsx',
  'src/components/manager/ga/common/filter/FilterButton.tsx',
  'src/components/manager/ga/common/filter/SortDropdown.tsx',
  'src/components/manager/ga/dashboard/ChartsSection.tsx',
  'src/components/manager/ga/dashboard/MemberStatsSection.tsx',
  'src/components/manager/ga/dashboard/StatCard.tsx',
  'src/components/manager/ga/dashboard/chart/CampaignRecruitmentChart.tsx',
  'src/components/manager/ga/dashboard/chart/ChannelMemberPieChart.tsx',
  'src/components/manager/ga/dashboard/chart/DeviceStatsChart.tsx',
  'src/components/manager/ga/dashboard/chart/MemberActivationDonutChart.tsx',
  'src/components/manager/ga/dashboard/chart/MemberTypeBarChart.tsx',
  'src/components/manager/ga/dashboard/chart/RejectionReportChart.tsx',
  'src/components/manager/ga/dashboard/section/AccessStatsSection.tsx',
  'src/components/manager/ga/dashboard/section/CampaignRecruitmentSection.tsx',
  'src/components/manager/ga/dashboard/section/CampaignSummarySection.tsx',
  'src/components/manager/ga/dashboard/section/ChannelMemberSection.tsx',
  'src/components/manager/ga/dashboard/section/DateFilterSection.tsx',
  'src/components/manager/ga/dashboard/section/DateRangePickerModal.tsx',
  'src/components/manager/ga/dashboard/section/MemberActivationSection.tsx',
  'src/components/manager/ga/dashboard/section/MemberTypeSection.tsx',
  'src/components/manager/ga/dashboard/section/RejectionReportSection.tsx',
  // data/manager_ga
  'src/data/manager_ga/common/filterOptions.ts',
  'src/data/manager_ga/community/categoriesData.ts',
  'src/data/manager_ga/community/postsData.ts',
  'src/data/manager_ga/member/blacklist.ts',
  'src/data/manager_ga/member/partners.ts',
  'src/data/manager_ga/member/reviewers.ts',
  'src/data/manager_ga/progress.ts',
  'src/data/manager_ga/rejected.ts',
  'src/data/manager_ga/reported.ts',
];

// 이모지 유니코드 범위 제거
function removeEmoji(str) {
  return str.replace(
    /[\u{1F300}-\u{1F5FF}\u{1F600}-\u{1F64F}\u{1F680}-\u{1F6FF}\u{1F700}-\u{1F77F}\u{1F780}-\u{1F7FF}\u{1F800}-\u{1F8FF}\u{1F900}-\u{1F9FF}\u{1FA00}-\u{1FA6F}\u{1FA70}-\u{1FAFF}\u{2702}-\u{27B0}\u{24C2}-\u{1F251}]+/gu,
    ''
  ).trim();
}

function fixHeader(content) {
  // 1. "주요 기능:" 섹션 제거 (섹션 + 항목들 + 후행 빈줄)
  //    패턴: " * 주요 기능:\n" 이후 " * - ..." 라인들 + " *\n" 빈줄
  content = content.replace(/ \* 주요 기능:\n( \* - [^\n]*\n)+ \*\n/g, '');
  content = content.replace(/ \* 주요 기능:\n( \* - [^\n]*\n)+/g, '');

  // 2. "사용 위치:" → "사용 페이지:" 통일
  content = content.replace(/ \* 사용 위치:\n/g, ' * 사용 페이지:\n');

  // 3. 헤더 구분선 제목의 이모지 제거
  //    패턴: /* ====\n   이모지 제목\n   ==== */
  content = content.replace(
    /(\/\* ={38,}\n\s+)([^\n]+)/g,
    (_, prefix, title) => prefix + removeEmoji(title)
  );

  // 4. JSDoc 내 연속 빈 줄 3개 이상 → 1개로
  content = content.replace(/( \*\n){3,}/g, ' *\n');

  return content;
}

let modified = 0;
const errors = [];

for (const rel of files) {
  const full = path.join(base, rel);
  try {
    const original = fs.readFileSync(full, 'utf-8');
    const fixed = fixHeader(original);
    if (fixed !== original) {
      fs.writeFileSync(full, fixed, 'utf-8');
      modified++;
      console.log(`✅ 수정: ${rel}`);
    } else {
      console.log(`⏭  변경 없음: ${rel}`);
    }
  } catch (e) {
    if (e.code === 'ENOENT') {
      errors.push(`❌ 파일 없음: ${rel}`);
    } else {
      errors.push(`❌ 오류 (${rel}): ${e.message}`);
    }
  }
}

console.log(`\n총 ${modified}개 파일 수정 완료`);
if (errors.length) {
  console.log('\n오류:');
  errors.forEach(e => console.log(e));
}
